core/src/main/java/com/linecorp/armeria/internal/logging/ContentPreviewingUtil.java

```diff
@@ -143,6 +164,27 @@ protected Throwable beforeError(Subscriber<? super HttpObject> subscriber, Throw
                 ctx.logBuilder().responseContentPreview(null);
                 return cause;
             }
+
+            @Override
+            protected void beforeCancel(Subscription subscription) {
+                produceResponseContentPreview(responseContentPreviewer);
+            }
+
+            private void produceResponseContentPreview(@Nullable ContentPreviewer responseContentPreviewer) {
```

Comment:
This method could be called twice if a `StreamMessage` subscribed with `NOTIFY_CANCELLATION` is canceled.

Reply:
Thanks, fixed. 😉